internal/service/tesla.go

```diff
@@ -0,0 +1,129 @@
+package service
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/DIMO-Network/shared/pkg/cipher"
+	"github.com/DIMO-Network/shared/pkg/db"
+	"github.com/DIMO-Network/tesla-oracle/internal/config"
+	dbmodels "github.com/DIMO-Network/tesla-oracle/models"
+	"github.com/rs/zerolog"
+	"regexp"
+	"strconv"
+)
+
+type TeslaService struct {
+	settings *config.Settings
+	logger   *zerolog.Logger
+	Cipher   cipher.Cipher
+	pdb      *db.Store
+}
+
+func NewTeslaService(settings *config.Settings, logger *zerolog.Logger, cipher cipher.Cipher, pdb *db.Store) *TeslaService {
+	return &TeslaService{
+		settings: settings,
+		logger:   logger,
+		Cipher:   cipher,
+		pdb:      pdb,
+	}
+}
+
+type VirtualKeyStatus int
+
+const (
+	Incapable VirtualKeyStatus = iota
+	Paired
+	Unpaired
+)
+
+// GetVehicleByVIN retrieves a vehicle by its VIN.
+func (ts *TeslaService) GetVehicleByVIN(ctx context.Context, logger *zerolog.Logger, pdb *db.Store, vin string) (*dbmodels.SyntheticDevice, error) {
+	tx, err := pdb.DBS().Writer.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelReadCommitted})
+	if err != nil {
+		logger.Error().Err(err).Msgf("Failed to begin transaction for vehicle %s", vin)
+		return nil, err
+	}
```

Comment:
nit: I don't think we need a transaction for a select since it doesn't have anything to commit or rollback.